kafkabp/config.go

```diff
@@ -105,7 +121,7 @@ func (cfg *ConsumerConfig) NewSaramaConfig() (*sarama.Config, error) {
 		return nil, ErrClientIDEmpty
 	}
 
-	var version sarama.KafkaVersion
+	version := defaultSaramaConfig.Version
```

Comment:
Rather than doing this, couldn't we just set `c.Version = version` in the if statement checking that a version is set?

Reply:
yea we can also do that. but I _slightly_ prefer this approach because:

1. We want to return all errors early, before constructing the actual sarama config, as much as possible.
2. With 1 in mind, this approach avoids a second `if cfg.Version != ""` block, so it's slightly better.